### PR TITLE
libffi Recipe

### PR DESCRIPTION
**Pull request by stefan-balke**

### CODE REVIEW

**Comment by stefan-balke** [comment]
FAIL: uniname/test-uninames.sh

That test failed in travis.

File: recipes/libffi/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+requirements:
+  build:
+    - autoconf
+    - automake
+    - libtool
+
+test:
+  commands:
+    - true
```
Comment (ocefpaf) [inline recipes/libffi/meta.yaml]: Some comment I made in the other recipes. (We have three easy merges waiting for a simple test :wink:)

**Comment by ocefpaf** [comment]
@stefan-balke when you have `skip: True # [win]` set try to add `[skip appveyor]` to your commit messages. I know that is annoying... The reason for that is because AppVeyor is very slow and puts all PRs in a (ever growing) queue and, since we are not building on Windows, we can safely remove this one from the queue.

**Comment by ocefpaf** [comment]
> + test -e /opt/conda/envs/_test/lib/libffi.a

It looks like the static library does not exist, was not build, or has another name on Linux. Do you have a Linux box to test that? If not let me know and I can do that for you.

**Comment by stefan-balke** [comment]
> Do you have a Linux box to test that?

Nope, no immediate access so would be cool if you could jump in.

**Comment by ocefpaf** [comment]
> Nope, no immediate access so would be cool if you could jump in.

On it... Hold on.

PS: In the future you can try our docker image for that. You would be testing exactly like the recipe will be build here.

**Comment by ocefpaf** [comment]
Conda puts everything on `$PREFIX/lib`, but those libraries are ending up in `$PREFIX/lib64`. I guess we can either move them to `$PREFIX/lib` or create that directory and put a soft ink in there. (Or better yet, patch the makefile to put them in `$PREFIX/lib`.)

@msarahan What do you recommend?

PS: @stefan-balke now you know why we want some tests :wink:

**Comment by ocefpaf** [comment]
BTW `--libdir=${PREFIX}/lib` does not work here :unamused:

**Comment by msarahan** [comment]
The assumption had always been that a single miniconda install is a single
architecture, so libraries should either be moved or copied to /lib, not
lib64.

I think that's important to stick with until someone comes up with a
compelling reason to have both arches in one miniconda install.  I'm not
convinced there is such a reason, as it is much easier to keep track of
with two miniconda installations of different arch.  Still, I'd be willing
to discuss use cases if anyone has one.

**Comment by ocefpaf** [comment]
Thanks for the clarification @msarahan. @stefan-balke can you add:

```shell
mkdir -p ${PREFIX}/lib
mv ${PREFIX}/lib64/* ${PREFIX}/lib/
```

at the end of your `build.sh`?

**Comment by ocefpaf** [comment]
Apologies @stefan-balke. Please add:

```shell
if [[ $(uname) == Linux ]]; then
  mkdir -p ${PREFIX}/lib
  mv ${PREFIX}/lib64/* ${PREFIX}/lib/
fi
```

**Comment by stefan-balke** [comment]
Done.

**Comment by ocefpaf** [comment]
Another one bites the dust! Thanks @stefan-balke!

**Comment by ocefpaf** [comment]
PS: I'd appreciate if you could open issues for Windows builds in the feedstocks. You might not get to it, but you never know if someone else will :wink:

**Pull request merged**